Src/IronPython/Compiler/Ast/TryStatement.cs

```diff
@@ -160,24 +165,35 @@ public class TryStatement : Statement {
                         )
                     );
             } else {
-                result = body;
+                // If the body throws an exception and there's no catch block,
+                // we should still update the current exception
+                result = AstUtils.Try(
+                    body
+                ).Catch(exception,
+                    Ast.Call(AstMethods.SetCurrentException, Parent.LocalContext, exception),
```

Comment:
Maybe I shouldn't be doing this here, and should instead do this over in the try/catch inside of `AddFinally()`